compiler/rustc_builtin_macros/src/deriving/from.rs

```diff
@@ -22,12 +22,23 @@ pub(crate) fn expand_deriving_from(
     push: &mut dyn FnMut(Annotatable),
     is_const: bool,
 ) {
-    let mut visitor = ExtractNonSingleFieldStruct { cx, field: None };
-    item.visit_with(&mut visitor);
+    let field = match &item {
+        Annotatable::Item(item) => match &item.kind {
+            ItemKind::Struct(_, _, data) => match data.fields() {
+                [field] => Some(field.clone()),
```

Comment:
Can you avoid the repeated `None`s by using a chained if-let?

Reply:
Ha, I keep forgetting that's a thing, since I didn't use them much yet. Thanks!